### PR TITLE
Feature/tcp websocket support

### PR DESCRIPTION
**Pull request by JR-Mitchell**
### What?

Implements websocket support

### Why?

I may want to use this framework in the future for multiplayer games

### Anything else?

This is kinda a minimal working example. A few nice-to-haves are missing, notably:
- Keeping track of client connection speeds using pong
- Checking that the host header is correct during handshake

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,25 +7,30 @@ A proof of concept / template for using Wasmoon and Teal to make a game that run
 - [npm](https://www.npmjs.com/)
 - [lua](https://www.lua.org/)
 - [luarocks](https://luarocks.org/)
+
+### Lua requirements (installed using luarocks)
 - [teal](https://github.com/teal-language/tl)
+- [cyan](https://github.com/teal-language/cyan)
```
Comment (fjebaker) [inline README.md]: I've not had the time to fully review this yet, but just glancing at this: do we really need all of these deps? I see the JS is using things like Babel and Webpack to bundle a really tiny static website. Could we, out of simplicity, drop some of these deps?

Same with `cyan`, is it _really_ needed?

Obviously this is not the PR to address this but to start the conversation.

Comment (JR-Mitchell) [reply README.md]: We probably could simplify the javascript side of things. As for `cyan`, I'm temporarily removing this dependency, as I can't seem to get the latest teal working for this project. But in theory, we would need it to compile the teal down to lua - unless we injected the teal runtime into the app.

It'd be a nice extra issue to create, to see how much we can streamline this.

Comment (fjebaker) [reply README.md]: Ah I see, I thought it was just some build system for automating things, not that it was crucial for the teal to lua steps.

**Comment by JR-Mitchell** [comment]
@fjebaker I've temporarily downgraded the teal dependency for this (waiting to hear back on https://github.com/teal-language/tl/issues/865), and put the updated requirement into the README.md.
I think you've raised a few good improvements:
- Try to streamline the build process
- Add documentation around how *I* use teal
Let me know if you can now successfully build, and feel free to make issues for these two things, so I can work on them separately from this PR

**Comment by JR-Mitchell** [comment]
Hey @fjebaker,
I've opened the following new issues:
- #4 
- #5 
- #6 
- #7 

If you can get this branch to run locally following the latest README.md instructions, please can you give this PR a (hopefully final) review - in this case, it closes #3 , and can be merged in.

**Comment by JR-Mitchell** [comment]
Eh I'm just merging this anyway

**Pull request merged**